modules/apps/asset/asset-display-page-service/src/main/java/com/liferay/asset/display/page/internal/upgrade/AssetDisplayPageServiceUpgrade.java

```diff
@@ -32,12 +35,20 @@
 	public void register(Registry registry) {
 		registry.register("0.0.1", "1.0.0", new DummyUpgradeStep());
 
-		registry.register("1.0.0", "1.1.0", new UpgradeAssetDisplayPage());
+		registry.register(
+			"1.0.0", "1.1.0",
+			new UpgradeAssetDisplayPage(_classNameLocalService));
 
 		registry.register(
 			"1.1.0", "2.0.0",
 			new BaseUpgradeSQLServerDatetime(
 				new Class<?>[] {AssetDisplayPageEntryTable.class}));
 	}
 
+	@Reference
+	private ClassNameLocalService _classNameLocalService;
+
+	@Reference
+	private JournalArticleLocalService _journalArticleLocalService;
```

Comment:
_journalArticleLocalService field declared and injected, but never used, furthermore, doing this we force asset-display-page-service module to depend on Journal service implementation, but the idea of new display pages is to make them asset-agnostic, available for any type of asset, independent from the modules we (don't)have in the Portal.

Reply:
This was required to make sure that AssetDisplayPage upgrade runs after JournalArticle's. If it ran before, the classNameIds did not match. However, I agree that we should not introduce this dependency and your suggestion in the next comment will solve it: https://github.com/jkappler/liferay-portal/pull/301#discussion_r236656061